modules/apps/dynamic-data-mapping/dynamic-data-mapping-expression/src/main/java/com/liferay/dynamic/data/mapping/expression/internal/DDMExpressionEvaluatorVisitor.java

```diff
@@ -565,46 +568,55 @@ protected String getFunctionName(Token functionNameToken) {
 		return (T)parseTree.accept(this);
 	}
 
-	private Optional<Method> _getApplyMethodOptional(
-		DDMExpressionFunction ddmExpressionFunction) {
+	private Method _getApplyMethod(
+			Class<?> ddmExpressionFunctionClass, int parametersTotal)
+		throws NoSuchMethodException {
 
-		List<Method> methods = Stream.of(
-			_getHierarchicalMethods(ddmExpressionFunction.getClass())
-		).filter(
-			method -> StringUtil.equals("apply", method.getName())
-		).collect(
-			Collectors.toList()
-		);
+		Class<?>[] classes = _getInterfaces(ddmExpressionFunctionClass);
 
-		Iterator<Method> iterator = methods.iterator();
+		Optional<Class<?>> classOptional = Stream.of(
+			classes
+		).filter(
+			clazz -> _isDDMExpressionFunctionNestedFunction(
+				clazz, parametersTotal)
+		).findFirst();
 
-		Method method = iterator.next();
+		if (classOptional.isPresent()) {
+			Class<?> clazz = classOptional.get();
 
-		Class<?>[] parameterTypes = method.getParameterTypes();
+			return clazz.getDeclaredMethods()[0];
+		}
 
-		Object object = new Object();
+		return ddmExpressionFunctionClass.getMethod("apply", Object[].class);
+	}
 
-		if ((parameterTypes.length == 1) &&
-			(parameterTypes[0] == object.getClass()) && iterator.hasNext()) {
+	private Class<?>[] _getInterfaces(Class<?> clazz) {
+		Set<Class<?>> classes = new HashSet<>();
 
-			return Optional.ofNullable(iterator.next());
+		if (clazz.getSuperclass() != null) {
+			Collections.addAll(classes, _getInterfaces(clazz.getSuperclass()));
 		}
 
-		return Optional.ofNullable(method);
+		Collections.addAll(classes, clazz.getInterfaces());
+
+		return classes.toArray(new Class<?>[0]);
 	}
 
-	private Method[] _getHierarchicalMethods(Class<?> clazz) {
-		Set<Method> methods = new HashSet<>();
+	private boolean _isDDMExpressionFunctionNestedFunction(
+		Class<?> clazz, int parametersTotal) {
 
-		if (clazz.getSuperclass() != null) {
-			Collections.addAll(
-				methods, _getHierarchicalMethods(clazz.getSuperclass()));
-		}
+		Class<?>[] classes = clazz.getInterfaces();
```

Comment:
Hi @rodrigopaulino. 
I noticed in the code that when `getInterfaces` is used, the variable is usually named as `interfaces` or `interfaceClasses`.
Do you think it makes sense to follow this pattern here?

Reply:
Hi, @carolmariaabb.

I'm following [this convention](https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1511752519/Java+conventions+in+Liferay#[inlineExtension]One-dimensional-collections-are-plurals) and I've also noticed there's a few ways in which people have declared this in the code (there's even "superClasses", which doesn't make sense at all 😂). So to me it's just dubious.
Regardless, I'll take your suggestion and use `interfaces` instead, because "interfaceClasses" doesn't make sense to me. 😛 

Thanks for asking.

Reply:
Hey, @carolmariaabb.

Just one more thing. Changing the variable to `interfaces` would be awkward when I'm streaming through the array, because `interface` is a reserved word and I can't call each item in the array that. Do you think it's still valid to rename it given this limitation?

Looking forward to hearing from you.

Reply:
Good point, @rodrigopaulino. Let's keep it the way it is, which also makes total sense to me.

Thanks for the link to the documentation.